### PR TITLE
Build with 3 JDK targets

### PR DESCRIPTION
**Pull request by yschimke**

### CODE REVIEW

**Comment by yschimke** [comment]
This wasn't as easy as I hoped, errors suggest its something funky with errorprone

https://travis-ci.org/square/okhttp/jobs/482051444

https://errorprone.info/docs/installation

```
14:50:13:869 [ERROR] /home/travis/build/square/okhttp/okhttp/src/main/java/okhttp3/internal/ws/WebSocketProtocol.java:[21,14] cannot access java.lang.Object
14:50:13:870 [ERROR]   bad class file: /modules/java.base/java/lang/Object.class
14:50:13:870 [ERROR]     class file has wrong version 55.0, should be 53.0
14:50:13:870 [ERROR]     Please remove or make sure it appears in the correct subdirectory of the classpath.
14:50:13:870 [ERROR] -> [Help 1]
```

**Comment by swankjesse** [comment]
Can we cut errorprone on that build or all of em? I think errorprone is great in application  but I can't recall anything it's caught in OkHttp lately.

**Comment by yschimke** [comment]
@swankjesse Let me test with remove it completely and then I'll add back via a profile

**Comment by yschimke** [comment]
- Fixed with error prone moved to a profile (needs review)
- javadoc fails currently on JDK 11 because summary on tables are not ok with html5
- jdk 11 failing with 

```
17:44:13:815 [INFO] Running okhttp3.UrlConnectionCacheTest
17:44:21:853 [ERROR] Tests run: 109, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 8.02 s <<< FAILURE! - in okhttp3.UrlConnectionCacheTest
17:44:21:856 [ERROR] varyAndHttps(okhttp3.UrlConnectionCacheTest)  Time elapsed: 0.013 s  <<< ERROR!
java.lang.UnsupportedOperationException: clientBuilder.sslSocketFactory(SSLSocketFactory) not supported on JDK 9+
	at okhttp3.UrlConnectionCacheTest.varyAndHttps(UrlConnectionCacheTest.java:1416)
17:44:21:858 [ERROR] secureResponseCaching(okhttp3.UrlConnectionCacheTest)  Time elapsed: 0.004 s  <<< ERROR!
java.lang.UnsupportedOperationException: clientBuilder.sslSocketFactory(SSLSocketFactory) not supported on JDK 9+
	at okhttp3.UrlConnectionCacheTest.secureResponseCaching(UrlConnectionCacheTest.java:267)
```

**Comment by yschimke** [comment]
@swankjesse I'll work on this over next week.  But it's more changes than I expected.  I think it's the right direction, get things confirmed building with all LTS releases, and best effort with 9, 10, 12.

**Comment by swankjesse** [comment]
Awesome.

File: .travis.yml [hunk 1]
```diff
@@ -2,14 +2,19 @@ language: java
...
-#  - openjdk11
+  - openjdk8
+  - openjdk11
 
-addons:
-  apt:
-    packages:
-      - oracle-java8-installer # Updates JDK 8 to the latest available.
+before_install:
+  - mvn -N io.takari:maven:wrapper -Dmaven=3.6.0
```
Comment (jaredsburrows) [inline .travis.yml]: Why use `mvnw`?

File: okhttp-logging-interceptor/src/test/java/okhttp3/logging/LoggingEventListenerTest.java [hunk 2]
```diff
@@ -31,6 +33,8 @@
 import org.junit.Rule;
 import org.junit.Test;
 
+import static java.util.Arrays.*;
```
Comment (jaredsburrows) [inline okhttp-logging-interceptor/src/test/java/okhttp3/logging/LoggingEventListenerTest.java]: Can you make sure not to have start imports?

File: .travis.yml (hunk 1 above)
Comment (yschimke) [reply .travis.yml]: For Java 11, using the most recent Maven and Maven Plugin versions is the most likely to be compatible.  This command is running with existing maven provided by travis.  It downloads Maven 3.6.0 so we have exact control over what to run with and installs it like the gradlew wrapper in the project.  Hence, mvnw after this point.

File: okhttp-tests/src/test/java/okhttp3/URLConnectionTest.java [hunk 3]
```diff
@@ -104,10 +104,13 @@
 import static okhttp3.mockwebserver.SocketPolicy.SHUTDOWN_OUTPUT_AT_END;
 import static okhttp3.mockwebserver.SocketPolicy.UPGRADE_TO_SSL_AT_END;
 import static okhttp3.tls.internal.TlsUtil.localhost;
+import static org.hamcrest.CoreMatchers.either;
```
Comment (swankjesse) [inline okhttp-tests/src/test/java/okhttp3/URLConnectionTest.java]: I'd like to migrate to AssertJ or Truth. We've resisted it to make ingesting OkHttp easier for AOSP; I think it's probably a bad trade off.

File: okhttp-tests/src/test/java/okhttp3/URLConnectionTest.java [hunk 4]
```diff
@@ -1180,16 +1185,18 @@ private void initResponseCache() {
   @Test public void disconnectDuringConnect_cookieJar() {
     final AtomicReference<HttpURLConnection> connectionHolder = new AtomicReference<>();
     class DisconnectingCookieJar implements CookieJar {
-      @Override public void saveFromResponse(HttpUrl url, List<Cookie> cookies) { }
+      @Override public void saveFromResponse(HttpUrl url, List<Cookie> cookies) {
+      }
```
Comment (swankjesse) [inline okhttp-tests/src/test/java/okhttp3/URLConnectionTest.java]: 👍🏻

**Comment by yschimke** [comment]
Just failed, taking a look.  https://travis-ci.org/square/okhttp/jobs/490938792

**Pull request merged**